### PR TITLE
perf: Disable round trip on write

### PR DESCRIPTION
**Pull request by maxsmythe**
This disables the round-tripping of cached data between serialized/deserialized JSON when caching in OPA. This reduces the amount of time a lock is held on OPA's cache, which helps mitigate the lock contention mentioned in
https://github.com/open-policy-agent/gatekeeper/issues/2060

Signed-off-by: Max Smythe <smythe@google.com>

### CODE REVIEW

**Comment by sozercan** [comment]
@maxsmythe looks like GK and unit tests are failing

**Comment by maxsmythe** [comment]
Yeah, turns out RoundTrip had a different signature than I thought. Hopefully the newest change works.

File: constraint/go.mod [hunk 1]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/golang/glog v1.0.0
 	github.com/google/go-cmp v0.5.9
-	github.com/open-policy-agent/opa v0.42.2
+	github.com/open-policy-agent/opa v0.44.0
```
Comment (ritazh) [inline constraint/go.mod:21]: This bump also fixes few CVEs. yay!

File: constraint/pkg/client/client.go [hunk 2]
```diff
@@ -415,7 +416,22 @@ func (c *Client) AddData(ctx context.Context, data interface{}) (*types.Response
 			}
 		}
 
-		err = c.driver.AddData(ctx, name, key, processedData)
+		// Round trip data to force untyped JSON, as drivers are not type-aware
+		bytes, err := json.Marshal(processedData)
```
Comment (ritazh) [inline constraint/pkg/client/client.go:420]: For my own knowledge, why does doing round trip here improves performance compare to in storage?

Comment (maxsmythe) [reply constraint/pkg/client/client.go:420]: Storage acquires a mutex, so a round trip there blocks all Rego evaluation. Round trips here can be done in parallel without blocking Rego evaluation.

TL;DR less CPU time spent with a mutex

**Review by ritazh — APPROVED** [review]
LGTM
